opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -349,6 +362,316 @@ def collect(
+            buckets._index_base = index
+
+        elif (
+            index < buckets._index_start
+            and (buckets._index_end - index) >= self._max_size
+        ):
+            is_rescaling_needed = True
+            low = index
+            high = buckets._index_end
+
+        elif (
+            index > buckets._index_end
+            and (index - buckets._index_start) >= self._max_size
+        ):
+            is_rescaling_needed = True
+            low = buckets._index_start
+            high = index
+
+        if is_rescaling_needed:
+
+            change = 0
+
+            while high - low >= self._max_size:
+                high = high >> 1
+                low = low >> 1
+
+                change += 1
+
+            if change == 0:
+                return
+
+            if change < 0:
+                raise Exception(f"Impossible change of scale: {change}")
+
+            new_scale = self._mapping.scale - change
+
+            self._positive.downscale(change)
+            self._negative.downscale(change)
+
+            if new_scale <= 0:
+                mapping = ExponentMapping(new_scale)
+            else:
+                mapping = LogarithmMapping(new_scale)
+
+            self._mapping = mapping
+
+            index = self._mapping.map_to_index(value)
+
+            if (
+                index < buckets._index_start
+                and (buckets._index_end - index) >= self._max_size
+            ) or (
+                index > buckets._index_end
+                and (index - buckets._index_start) >= self._max_size
+            ):
+                raise Exception("Downscaling logic error")
+
+        if index < buckets._index_start:
+            span = buckets._index_end - index
+
+            if span >= len(buckets.counts):
+                buckets.grow(span + 1, self._max_size)
+
+            buckets._index_start = index
+
+        elif index > buckets._index_end:
+            span = index - buckets._index_start
+
+            if span >= len(buckets.counts):
+                buckets.grow(span + 1, self._max_size)
+
+            buckets._index_end = index
+
+        bucket_index = index - buckets._index_base
+
+        if bucket_index < 0:
+            bucket_index += len(buckets.counts)
+
+        buckets.increment_bucket(bucket_index)
+
+    def collect(
+        self,
+        aggregation_temporality: AggregationTemporality,
+        collection_start_nano: int,
+    ) -> Optional[_DataPointVarT]:
+        """
+        Atomically return a point for the current value of the metric.
+        """
+
+        with self._lock:
+            if not any(self._negative.counts) and not any(
+                self._positive.counts
+            ):
+                return None
+
+            start_time_unix_nano = self._start_time_unix_nano
+            sum_ = self._sum
+            max_ = self._max
+            min_ = self._min
+
+            self._negative._counts = [0]
+            self._positive._counts = [0]
```

Comment:
Should the _index_start, _index_end, _index_base fields be updated to match this reset?

Reply:
Right, fixed.